tests-web/www/js/controls.js

```diff
@@ -105,21 +115,80 @@ export class LayerGroup {
       };
       entry.appendChild(download);
       members.push(entry);
-
-      // This is maybe an odd time to sync this state
-      if (layer.enabled) {
-        this.map.addLayer(layer.layer);
-      } else {
-        this.map.removeLayer(layer.layer);
-      }
     }
     var div = makeDiv(members);
     div.id = this.name;
     return div;
   }
+
+  remove() {
+    for (const layer of this.layers) {
+      this.map.removeLayer(layer.layer);
+    }
+  }
 }
 
-// Manages a list of groups
+// Contains a bunch of LayerGroups, displays exactly one at a time, and can
+// scroll through the sequence.
+export class SequentialLayerGroup {
+  constructor(name, groups) {
+    this.name = name;
+    this.groups = groups;
+    this.current = 0;
+
+    // Start with only the first enabled
+    for (var i = 0; i < this.groups.length; i++) {
+      this.groups[i].setEnabled(i == this.current);
+    }
+  }
+
+  renderControls() {
+    const prev = makeButton("Previous");
+    prev.disabled = this.current == 0;
+    prev.onclick = () => {
+      this.groups[this.current].setEnabled(false);
+      this.current -= 1;
+      this.groups[this.current].setEnabled(true);
+
+      // Rerender
+      document.getElementById(this.name).replaceWith(this.renderControls());
+    };
+
+    const label = document.createTextNode(
+      `${this.current + 1} / ${this.groups.length}`
+    );
+
+    const next = makeButton("Next");
+    next.disabled = this.current == this.groups.length - 1;
+    next.onclick = () => {
+      this.groups[this.current].setEnabled(false);
+      this.current += 1;
+      this.groups[this.current].setEnabled(true);
+
+      // Rerender
+      document.getElementById(this.name).replaceWith(this.renderControls());
+    };
+
+    var row1 = L.DomUtil.create("u");
```

Comment:
I was looking for a way to do something like `const row1 = document.makeDOMElement(`<u>${this.name}</u>`)` to create elements more succinctly. But I couldn't spot a simple answer for this. https://stackoverflow.com/questions/494143/creating-a-new-dom-element-from-an-html-string-using-built-in-dom-methods-or-pro has a bunch of ways, and maybe `new DOMParser().parseFromString(html, 'text/html').body.childNodes[0]` is the very verbose one that's most modern?

Reply:
I think the easiest way is to assign to innerHTML. This sort of thing:

```js
document.getElementById(blah).innerHTML = `<${tag} class="${classnames(...)}">Do it</${tag}>`;
```

You can create elements incrementally using the [low level API](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement#basic_example):

```js
function addElement () {
  const newDiv = document.createElement("div");
  const newContent = document.createTextNode("Hi there and greetings!");
  newDiv.appendChild(newContent);
  const currentDiv = document.getElementById("div1");
  document.body.insertBefore(newDiv, currentDiv);
}
```

Reply:
There has to be an element out there somewhere to do `innerHTML`. So in these cases where we're building up new things to insert somewhere, we could insert a blank element first, then overwrite `innerHTML` for it?